src/com/gfredericks/test/chuck/generators.clj

```diff
@@ -252,3 +255,60 @@
   (gen/fmap (fn [ks]
               (select-keys m ks))
             (subsequence (keys m))))
+
+(def ^:private valid-offset-fns #{ct/millis ct/seconds ct/minutes ct/hours ct/days ct/months ct/years})
+
+(defn gen-datetime
+  "Generates datetime within given range and format.
+
+   base-datetime => By default it'll calculate the dates from year 2000.
+                    Generally this is a good idea instead of using (ct/now)
+                    since giving the same seed will generate the same output.
+                    If you would like to generate from a differnt base-datetime,
+                    Pass for example (ct/now) to use current time,
+                    Or pass a specific date-time (ct/date-time 2011 11 24)
+
+   offset-min & offset-max => The offset number range
+                              By default it is -1000 to 1000
+
+   offset-fns => List of functions which will be used with the given offset.
+                 It randomly picks one of the functions and
+                 applies the random offset with the given range.
+                 Check valid-offset-fns for possible values.
+                 By default its all the values of valid-offset-fns.
+
+   format-fn => Function to format the DateTime obj into.
+                By default it is identity that is return DateTime Object
+                Other valid formatters are:
+                clj-time.core/to-long, clj-time.core/to-epoch etc
+                Check clj-time.core/to-* functions for valid formatters.
+
+   For example If you would like to generate datetime
+   from last 10 months to next 10 months:
+   (gen/sample (gen-datetime {:offset-fns [clj-time.core/months]
+                              :offset-min -10
+                              :offset-max 10}))
+   =>
+   (#<DateTime 1999-11-01T00:00:00.000Z>
+    #<DateTime 1999-12-01T00:00:00.000Z>
+    #<DateTime 2000-05-01T00:00:00.000Z>
+    ....)"
+  ([]
+   (gen-datetime {}))
+  ([{:keys [base-datetime offset-fns offset-min offset-max format-fn]
+     :or {offset-fns valid-offset-fns
+          offset-min -1000
+          offset-max 1000
+          base-datetime (ct/date-time 2000)
+          format-fn identity}}]
+   {:pre [(<= offset-min offset-max)
+          (cset/subset? offset-fns
+                        valid-offset-fns)]}
```

Comment:
does this work when `offset-fns` is not a set? if not an easy alternative is `(every? valid-offset-fns offset-fns)`, which also lets you remove the `cset` require.

In any case it might be better for the default of `offset-fns` to be an ordered collection, because `gen/elements` shrinks to earlier elements, so perhaps we should e.g. shrink from `years` to `millis`. Which might be happening currently due to some implementation detail about how the set on line `259` is ordered, but being explicit would be better.